### ISSUE TITLE
Fieldsets in sub-directories

### ISSUE THREAD

**Issue by jacksleight** [statamic/cms#3178]
## Bug Description

I'm not sure if this is a bug or a side-effect of something that's not supported but almost works by accident.

If you create a fieldset in a subdirectory, such as `/resources/fieldsets/components/image.yaml`, it appears in the UI with the handle `components.image`, and you can use it elsewhere correctly. However:

1. Although the fields function perfectly in the CP, on the frontend the values are not converted to Value objects
2. If you try to edit it in the CP it gets saved to a different location: `/resources/fieldsets/components.image.yaml`
3. it does not appear to be possible to create a subdirectory fieldset in the CP, as the `.` and `/` characters are invalid

I realise this is probably just an unsupported feature, but as the CP does almost handle it correctly I thought it could be a bug. If it is unsupported, it would be a very useful feature to add.

### Values issue

1. Create a `/resources/fieldsets/components/image.yaml` file
2. Use it in a Bard field
3. Check the field values on the frontend, they're not Value objects

### File issue

1. Create a `/resources/fieldsets/components/image.yaml` file
2. Go into the CP and edit and save it
3. Check for `/resources/fieldsets/components.image.yaml` file

![Screenshot 2021-01-30 at 16 25 36](https://user-images.githubusercontent.com/126740/106361905-d5f60580-6317-11eb-8a3e-9014485adc0f.png)
![Screenshot 2021-01-30 at 16 25 53](https://user-images.githubusercontent.com/126740/106361908-d7bfc900-6317-11eb-80e6-49edfcc8a31e.png)

## Environment

Statamic 3.0.40 Solo
Laravel 8.25.0
PHP 7.3.16
local/extensions dev-main

**Install method (choose one):**
- Existing Laravel app

**Comment by damcclean** [comment]
Yeah, it sounds like fieldsets in sub directories aren't supported yet. However, it's probably not a massive effort to actually add support for it, by the sounds of it.

**Comment by aerni** [comment]
I also found, that the `find` and `exists` methods in `Statamic\Fields\FieldsetRepository` will actually look for a subfolder path. The following will only return true if the fieldset exists at `fieldsets/components/image.yaml`. It will return false if it's `fieldsets/components.image.yaml`.

```php
Fieldset::setDirectory(resource_path('fieldsets'))->exists('components.image');
```

**Comment by duncanmcclean** [comment]
Will be possible in the next alpha of Statamic 6! 🚀
